GVRf/Framework/src/org/gearvrf/utility/Preconditions.java

```
@@ -0,0 +1,103 @@
+                    dataMessage);
+        }
+    }
+    
+    /**
+     * We can see that there is a wide use of key-value pair for the parameters which are used in shader.
+     * This function is to ensure that the key is meaningful.
+     * @param key
+     *          Key in the key-value pair.
+     *          
+     * @throws IllegalArgumentException if the key is null or empty.
+     */
+    public static void checkKeyNotNullOrEmpty(String key){
+        if(key == null || key.trim().isEmpty()){
+            throw Exceptions.IllegalArgument("Current key is %s which is not a proper key.", key == null ? "null" : "empty");
+        }
+    }
+
+    /**
+     * In GVRF, there are a lot of arrays uses with expected number of components inside.
+     * This function is to ensure that the input array is legal according to the expected components' number. 
+     * Otherwise it will throw an IllegalArgumentException.
+     * @param dataLength
+     *          Length of array data.
+     * @param expectedComponents
+     *          Expected number of components.
+     * @throws IllegalArgumentException if the dataLength is not divisible by the expected number.
+     */         
+    public static void checkDivisibleDataLength(int dataLength, int expectedComponents) {
+        if (dataLength == 0 || dataLength % expectedComponents != 0) {
+            throw Exceptions
+                    .IllegalArgument(
+                            "The input array should be an array of %d-component elements whose length is non-zero and divisible by %d. But current data length is %d.",
+                            expectedComponents, expectedComponents, dataLength);
+        }
+    }
+    
+    /**
+     * In common shader cases, NaN makes little sense. Correspondingly, GVRF is going to use Float.NaN as illegal 
+     * flag in many cases. Therefore, we need a function to check if there is any setX that is using NaN as input.
```

The reason of this is for your fix on getFloat(). Currently your pull request is going to use the GVRJniException. I talked to Tom that he also thinks Float.NaN should be an illegal parameter for shader at any case. So that if we use Float.NaN as an illegal identifier in getFloat(). Then we can throw a Java Exception when getFloat() is doing something that is not correct.